pkg/preparer/orchestrate_test.go

```diff
@@ -360,3 +366,22 @@ func TestPreparerWillAcceptSignatureWhenKeyringIsNil(t *testing.T) {
 	p.keyring = nil
 	Assert(t).IsTrue(p.verifySignature(*manifest, logging.DefaultLogger), "expected the preparer to verify the signature when no keyring given")
 }
+
+func TestPreparerWillNotLaunchAnAppAsRoot(t *testing.T) {
+	illegalManifest := pods.Manifest{Id: "foo", RunAs: "root"}
+
+	p, hooks, _ := testPreparer(t, &FakeStore{})
+	podChan := make(chan pods.Manifest)
+	quitChan := make(chan struct{})
+	go p.handlePods(podChan, quitChan)
+
+	podChan <- illegalManifest
+
+	// wait a second to see if auth fail hook is run. Unfortunately can't assert that pod wasn't installed
+	// because the pod is created inside p.handlePods()
```

Comment:
I can't get a handle on the pod to make sure installation didn't happen because it is created within `p.handlePods`. I'm unable to test that the preparer is allowed to run as root either without making the test take at least a second for the run after auth hook to NOT be called